### ISSUE TITLE
Backpropagation of training RPN and Mask RCNN at same time

### ISSUE THREAD

**Issue by stillwaterman** [matterport/Mask_RCNN#1123]
Hello, I have a question about the backpropagation in training. In the original paper, for convenient ablation, RPN is trained separately and does not share features with Mask R-CNN, unless specified. But I see the demo code of training train RPN and Mask R-CNN at same time.  During the training(code version), I do not know if rpn_class_loss, rpn_bbox_loss only affect the layers of RPN and backbone, they have no effect about the part of Mask R-CNN layers? We know the loss value that will be minimized by the model will then be the sum of all individual losses in Keras. Does the 'big loss' work on all layers in the network or just the backbone? I want to know whether the different training methods will affect the model and I really want to know the effective range all kinds of loss. Could you help me to figure it out? 
Looking forward to your reply!!

**Comment by stillwaterman** [comment]
@stygian2a, hello, I saw you answered some questions about loss function, could you help me solve some questions above?

**Comment by cad0p** [comment]
Hi @stillwaterman I see you closed this issue, but no response to your question is here!

**Comment by fnitiwat** [comment]
i looking for this question too. 
Is there anyone who can help answer this question?
